front-end/src/app/reports/f3x/create-workflow/create-f3x-step2.component.ts

```diff
@@ -0,0 +1,115 @@
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { mergeMap, Observable, Subject, takeUntil, skipUntil } from 'rxjs';
+import { Store } from '@ngrx/store';
+import { refreshCommitteeAccountDetailsAction } from '../../../store/committee-account.actions';
+import { CommitteeAccount } from 'app/shared/models/committee-account.model';
+import { selectCommitteeAccount } from 'app/store/committee-account.selectors';
+import { LabelUtils, PrimeOptions, StatesCodeLabels, CountryCodeLabels } from 'app/shared/utils/label.utils';
+import { ValidateService } from 'app/shared/services/validate.service';
+import { schema as f3xSchema } from 'fecfile-validate/fecfile_validate_js/dist/F3X';
+import { F3xSummary } from 'app/shared/models/f3x-summary.model';
+import { F3xSummaryService } from 'app/shared/services/f3x-summary.service';
+
+@Component({
+  selector: 'app-create-f3x-step2',
+  templateUrl: './create-f3x-step2.component.html',
+  styleUrls: ['./create-f3x-step2.component.scss'],
+})
+export class CreateF3xStep2Component implements OnInit, OnDestroy {
+  report: F3xSummary | null = null;
+  formProperties: string[] = [
+    'change_of_address',
+    'street_1',
+    'street_2',
+    'city',
+    'state',
+    'zip',
+    'memo_checkbox',
+    'memo',
+  ];
+  stateOptions: PrimeOptions = [];
+  countryOptions: PrimeOptions = [];
+  formSubmitted = false;
+  destroy$: Subject<boolean> = new Subject<boolean>();
+  committeeAccount: CommitteeAccount | null = null;
+  committeeAccount$: Observable<CommitteeAccount> | null = null;
+
+  form: FormGroup = this.fb.group(this.validateService.getFormGroupFields(this.formProperties));
+
+  constructor(
+    private router: Router,
+    private route: ActivatedRoute,
+    private reportService: F3xSummaryService,
+    private validateService: ValidateService,
+    private fb: FormBuilder,
+    private store: Store
+  ) {}
+
+  ngOnInit(): void {
+    // Refresh committee account details whenever page loads
+    this.store.dispatch(refreshCommitteeAccountDetailsAction());
+
+    this.committeeAccount$ = this.store.select(selectCommitteeAccount);
+    this.committeeAccount$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((committeeAccount: CommitteeAccount) => (this.committeeAccount = committeeAccount));
+    this.stateOptions = LabelUtils.getPrimeOptions(StatesCodeLabels);
+    this.countryOptions = LabelUtils.getPrimeOptions(CountryCodeLabels);
+
+    this.route.paramMap
+      .pipe(
+        takeUntil(this.destroy$),
+        skipUntil(this.committeeAccount$),
+        mergeMap((params): Observable<F3xSummary> => {
+          const id = Number(params.get('id'));
+          return this.reportService.get(id);
+        })
+      )
+      .subscribe((report: F3xSummary) => {
+        this.report = report;
+        this.form.patchValue({
+          change_of_address: 'not-selected',
+          street_1: this.report.street_1 ? this.report.street_1 : this.committeeAccount?.street_1,
```

Comment:
if you combine the committeeAccount and parammap observables, you dont need to create a `committeeAccount` component property.  combineLatest gives you the functionality of skipUntil and lines 37 and 55-57 all in one call

Reply:
Yeah, I like that. We need to wrap that code up into an ng resolver to port it across pages before load